goosebit/settings/schema.py

```diff
@@ -0,0 +1,74 @@
+from pathlib import Path
+from typing import Annotated, Iterable
+
+from pydantic import BaseModel, BeforeValidator, Field
+from pydantic_settings import (
+    BaseSettings,
+    PydanticBaseSettingsSource,
+    SettingsConfigDict,
+    YamlConfigSettingsSource,
+)
+
+from goosebit.permissions import Permissions, PermissionsBase
+
+from .const import BASE_DIR, LOGGING_DEFAULT, PWD_CXT
+
+
+def parse_permissions(items: Iterable[str]):
+    permissions = set()
+    for p in items:
+        permissions.update(Permissions.from_str(p))
+    return permissions
+
+
+class User(BaseModel):
+    username: str
+    hashed_pwd: Annotated[str, BeforeValidator(PWD_CXT.hash)] = Field(validation_alias="password")
+    permissions: Annotated[set[PermissionsBase], BeforeValidator(parse_permissions)]
+
+    def get_json_permissions(self):
+        return [str(p) for p in self.permissions]
+
+
+class PrometheusSettings(BaseModel):
+    port: int = 9090
+
+
+class MetricsSettings(BaseModel):
+    prometheus: PrometheusSettings = PrometheusSettings()
+
+
+class GooseBitSettings(BaseSettings):
+    model_config = SettingsConfigDict(env_prefix="goosebit_")
```

Comment:
```suggestion
    model_config = SettingsConfigDict(env_prefix="GOOSEBIT_")
```

Even though this shouldn't make a difference, it might for the reader. Envronment variables should be ALL_CAPS.

Reply:
Environment variables when parsed with pydantic are case-insensitive, we can change this but it will definitely not make any difference.

Reply:
As I said, it is only for the readers of the code.